src/main/java/edu/harvard/iq/dataverse/EditDatafilesPage.java

```diff
@@ -1377,6 +1379,12 @@ public String save() {
                         dataset = datasetService.find(dataset.getId());
                     }
                 }
+                //We also need to reindex since no commands that include reindexing have been called.
+                indexService.indexDataset(dataset, true);
```

Comment:
To fix the bug in 4.15.1, I could have added this at line ~1430, but reindexing is also needed at ~1438 for another case (see issue). I think adding it here catches both cases and avoids reindexing when it isn't needed (as would be the case if 1438 were uncommented).

I also tried to mimic the new exception handling/logging from #4425.

Reply:
So, really this logic *should* go through a Command. Not sure how it got in without that. We'll need to decide if we want his simple fix or require the refactoring to a command.

(my guess is we'll seetle on accepting this but open up a new issue for the refactoring - being in a command is important for permissions, automatic logging, and allowing same code for UI and API.

Reply:
+1 - it might be as easy as always forcing datasetUpdateRequired == true  as was done with prov around line 1178 (or just removing that flag completely).